File: shield.rb
# frozen_string_literal: true
require_relative 'dice'

module Irrgarten
  class Shield
    def initialize(protection, uses)
      @protection = protection
      @uses = uses
    end

    def protect

      if @uses > 0
        @uses -=1
        return @protection
      end
      0
    end

    def to_s
      "S[" + @protection.to_s + "," + @uses.to_s + "]"
    end

    def discard
      rand = Dice.new
      rand.discard_element(@uses)
    end

  end
end

File: Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author gonzaalovd
 */
public class Shield extends CombatElement{
    
    public Shield(float protection, int uses){
        super(protection, uses);
    }
    
    public float protect(){
        return this.produceEffect();
    }
    
    
    public String toString(){
        return "S" + super.toString();
    }
    
}
